tests/test_harmonic_mean.py

```diff
@@ -56,3 +56,28 @@ def test__compute_hm(start_idx, end_idx, expected_hm):
     x = partdf[SP_SOUND_SPEED]
     H = w.dropna().sum() / (w / x).dropna().sum()
     assert result_hm == round(H, 3)
+
+
+# Add test for missing columns in dataframe
+def test__compute_hm_missing_columns():
+    svdf = pd.DataFrame({"random_column1": [1, 2, 3], "random_column2": [4, 5, 6]})
+    start_depth = 0
+    end_depth = 2
+    with pytest.raises(ValueError):
+        _compute_hm(svdf, start_depth, end_depth)
+
+
+# Add test for empty dataframe
+# def test_sv_harmonic_mean_empty_dataframe():
+#     svdf = pd.DataFrame()
+#     start_depth = 0
+#     end_depth = 2
+#     with pytest.raises(ValueError):
+#         sv_harmonic_mean(svdf, start_depth, end_depth)
```

Comment:
Could you remove the commented function? Thanks!

Reply:
Done! Thanks for the review! 
I will add the check in the sv_harmonic_mean if the DataFrame is empty before trying to access its columns and the corresponding test case in the next commit.
```python
def sv_harmonic_mean(svdf, start_depth, end_depth):
    if svdf.empty:
        raise ValueError("Dataframe is empty! Please check your data inputs.")
    svdf = svdf[svdf[SP_SOUND_SPEED] > 0].reset_index(drop=True)
    # the rest of the function...
```

Reply:
Let's open a new Pull Request with the changes you mentioned above.